### PR TITLE
more complex querys by jpa

### PR DESCRIPTION
**Pull request by joredudiaz**
## Summary

- find Book By Publisher And Year And Genre
- find User By Birth Date And Name Containing IgnoreCase
- Junit test

## Jira Story

https://trello.com/c/TmlVFPxK

### CODE REVIEW

File: src/main/java/wolox/training/repositories/UserRepository.java [hunk 1]
```diff
@@ -28,4 +29,14 @@
...
 
+    /**
+     * Find users with birth date between two dates and name contains certain characteres
+     *
+     * @param startDate
+     * @param endDate
+     * @param name
+     * @return User[]
+     */
+    User[] findByBirthDateBetweenAndNameContainingIgnoreCase(LocalDate startDate, LocalDate endDate,
```
Comment (darkBasic666) [inline src/main/java/wolox/training/repositories/UserRepository.java]: Use Optional

**Review by darkBasic666 — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/repositories/BookRepository.java [hunk 2]
```diff
@@ -28,4 +28,14 @@
...
 
+    /**
+     * Find a book by publisher, year and genre
+     *
+     * @param publisher book's publisher
+     * @param year      book's year
+     * @param genre     book's genre
+     * @return Book[]
+     */
+    Book[] findByPublisherAndYearAndGenre(String publisher, String year, String genre);
```
Comment (Jimenarosello) [inline src/main/java/wolox/training/repositories/BookRepository.java]: Why are you using Book[] and User[] rather than Lists? 💭

**Review by Jimenarosello — CHANGES_REQUESTED** [review]
Just one comment out of curiosity 😅

File: src/main/java/wolox/training/repositories/BookRepository.java (hunk 2 above)
Comment (joredudiaz) [reply src/main/java/wolox/training/repositories/BookRepository.java]: jajjja emmm I forgot every time

**Review by darkBasic666 — APPROVED** [review]

**Pull request merged**